drivers/sensor/tdk/icm42688/icm42688_decoder.c

```diff
@@ -445,12 +448,19 @@ static int icm42688_fifo_decode(const uint8_t *buffer, struct sensor_chan_spec c
 
 			data->readings[count].timestamp_delta = ts_delta;
 
-			rc = icm42688_read_imu_from_packet(buffer, true, edata->header.accel_fs, 0,
-							   &data->readings[count].x);
-			rc |= icm42688_read_imu_from_packet(buffer, true, edata->header.accel_fs, 1,
-							    &data->readings[count].y);
-			rc |= icm42688_read_imu_from_packet(buffer, true, edata->header.accel_fs, 2,
-							    &data->readings[count].z);
+			q31_t reading[3];
+
+			for (int i = 0; i < 3; i++) {
+				rc = icm42688_read_imu_from_packet(
```

Comment:
you're not checking / overriding rc in each iteration

Reply:
Good catch!